### DISCUSSION TITLE
Cached Data in useInfiniteQuery Becomes Null After Several Hours Despite gcTime and staleTime Set to Infinity, is my setup wrong?

### DISCUSSION THREAD

**Discussion by JCs3lma** [TanStack/query#8177] [Q&A]
I'm using useInfiniteQuery in my Next.js project with TanStack Query (@tanstack/react-query version 5.55.4) and encountered an issue where the cached data disappears after several hours of inactivity, even though both gcTime and staleTime are set to Infinity.

Details:

1. When I initially reload the page, the data displays correctly.
2. I leave the page inactive for a few hours.
3. Upon returning, I notice that the data is no longer being displayed, and when I check React DevTools, the query key is still present, but the data is null.

Query Configuration:
In my code, I’m using useInfiniteQuery with the following configuration:
```
export const useSearchSchoolsFn = ({
  countryName,
  cityName,
  schoolName,
  filters,
  accessToken,
  take,
  sortType,
  courseAge,
  accommodationTypeSelected,
  roomTypeSelected,
  mealTypeSelected,
  locale,
  isSearchPulling = false,
}) => {
  return useInfiniteQuery({
    queryKey: [
      'searchSchools',
      countryName,
      cityName,
      schoolName,
      filters,
      sortType,
      accommodationTypeSelected,
      roomTypeSelected,
      mealTypeSelected,
      courseAge,
    ],
    queryFn: ({ pageParam }: { pageParam: number }) =>
      searchSchools(
        countryName,
        cityName,
        schoolName,
        filters,
        accessToken,
        take,
        sortType,
        courseAge,
        accommodationTypeSelected,
        roomTypeSelected,
        mealTypeSelected,
        locale,
        {
          pageParam: pageParam,
        }
      ),
    initialPageParam: 10,
    gcTime: Infinity,
    staleTime: Infinity,
    refetchOnReconnect: true,
    refetchOnWindowFocus: true,
    enabled: isSearchPulling,
    getNextPageParam: (lastPage) => lastPage.nextTake,
  });
};
```

Component Code:
I also have the following component-level state and use the useSearchSchoolsFn query:
```
const [page, setPage] = useState(0);
const [pulledPages, setPulledPages] = useState(0);
const [searchTake, setSearchTake] = useState(defaultTake);
const [searchCurrentTake, setSearchCurrentTake] = useState(defaultTake);
const [isSearchPulling, setIsSearchPulling] = useState(false);
const [searchCountryName, setSearchCountryName] = useState('');
const [searchCityName, setSearchCityName] = useState('');
const [searchSchoolName, setSearchSchoolName] = useState('');
const [searchFilters, setSearchFilters] = useState<any>([]);
const [searchCourseAge, setSearchCourseAge] = useState('');
const [searchRoomType, setSearchRoomType] = useState([]);
const [searchMealType, setSearchMealType] = useState([]);
const [searchAccommodationTypeSelected, setSearchAccommodationTypeSelected] = useState([]);

const {
  data: schoolData,
  status: searchStatus,
  isFetchingNextPage: isSearchFetchingNextPage,
  fetchNextPage,
  hasNextPage: hasMorePage,
} = useSearchSchoolsFn({
  countryName: searchCountryName,
  cityName: searchCityName,
  schoolName: searchSchoolName,
  filters: searchFilters,
  accessToken: accessToken,
  take: searchTake,
  sortType: sortType,
  courseAge: searchCourseAge,
  accommodationTypeSelected: searchAccommodationTypeSelected,
  roomTypeSelected: searchRoomType,
  mealTypeSelected: searchMealType,
  isSearchPulling: isSearchPulling,
  locale: app.lang,
});
```
Expected Behavior:
Since I’ve set both gcTime and staleTime to Infinity, I expect the cached data to persist indefinitely and not be removed or become null after a few hours of inactivity.

Questions:

- Is there any reason why the data might still be getting cleared or refetched after being inactive for a while, despite the Infinity settings?
- Could there be another setting or issue (e.g., query key changes, invalidation, or browser inactivity) that is causing the cached data to be dropped?

Any help or insights would be greatly appreciated. Thank you!

**Comment by TkDodo** [comment]
> Is there any reason why the data might still be getting cleared or refetched after being inactive for a while, despite the Infinity settings?

no.

> Could there be another setting or issue (e.g., query key changes, invalidation, or browser inactivity) that is causing the cached data to be dropped?

don’t think so.

I would need to see it (as in: a minimal reproduction)
